src/pytest_bdd/parser.py

```diff
@@ -318,9 +319,11 @@ def add_line(self, line: str) -> None:
         :param str line: Line of text - the continuation of the step name.
         """
         self.lines.append(line)
+        if "full_name" in self.__dict__:
+            del self.__dict__["full_name"]
```

Comment:
```suggestion
            del self.full_name
```

Reply:
I did try that first but it broke all the tests.

From https://stackoverflow.com/questions/59899732/python-cached-property-how-to-delete

"Your code will try to delete the method itself, and not the cached value. The correct way to do it is to delete the key from self.__dict__ (which is where the data is actually stored)."

Reply:
fixed it